noun/events.c

```diff
@@ -7,7 +7,7 @@
 
 #include "all.h"
 
-#if 0
+#ifdef U3_SNAPSHOT_VALIDATION
```

Comment:
Should we add a build option and `#mesondefine` for `U3_SNAPSHOT_VALIDATION`?

Reply:
Keeping around all these `#if 0` blocks is definitely an anti-pattern, but I'm not sure what the right thing is here. Is this something that we want to be able to turn on sometimes, or is it just deadish code?

Reply:
I'm not sure. There's a lot of this sort of verification and debugging code around. It's probably enough to gate it with a specific `#define`, we can always add options later if we end up using this stuff. And if we never end up using it, we'll eventually just want to remove it.